app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+                Icon(Icons.Filled.ArrowBack)
+            }
+            Text(
+                text = plantName,
+                color = MaterialTheme.colors.onSurface,
+                style = MaterialTheme.typography.h6,
+                modifier = Modifier.weight(1f).fillMaxSize()
+                    .wrapContentSize(Alignment.Center)
+            )
+            val shareAccessibilityLabel = stringResource(R.string.menu_item_share_plant)
+            IconButton(
+                onShareClicked,
+                Modifier.gravity(Alignment.CenterVertically).semantics {
+                    accessibilityLabel = shareAccessibilityLabel
+                }
+            ) {
+                Icon(Icons.Filled.Share)
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImageHeader(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    onFabClicked: () -> Unit,
+    isPlanted: Boolean,
+    modifier: Modifier = Modifier,
+    transitionModifier: Modifier = Modifier
+) {
+    val imageHeight = state { 0 }
+
+    Stack(modifier.fillMaxWidth()) {
+        PlantImage(scrollerPosition, imageUrl, transitionModifier.onPositioned {
+            imageHeight.value = it.size.height
+        })
+        if (!isPlanted) {
+            val fabModifier = if (imageHeight.value != 0) {
+                Modifier
+                    .gravity(Alignment.TopEnd)
+                    .padding(end = 8.dp)
+                    .offset(y = getFabOffset(imageHeight.value, scrollerPosition))
+                    .plus(transitionModifier)
+            } else {
+                Modifier
+            }
+            val fabAccessibilityLabel = stringResource(R.string.add_plant)
+            FloatingActionButton(
+                onClick = onFabClicked,
+                shape = SunflowerFabShape,
+                modifier = fabModifier.semantics {
+                    accessibilityLabel = fabAccessibilityLabel
+                }
+            ) {
+                Icon(Icons.Filled.Add)
+            }
+        }
+    }
+}
+
+@Composable
+private fun PlantImage(
+    scrollerPosition: ScrollerPosition,
+    imageUrl: String,
+    modifier: Modifier = Modifier
+) {
+    ParallaxEffect(scrollerPosition, modifier) { parallaxModifier ->
+        CoilImageWithCrossfade(
+            data = imageUrl,
+            contentScale = ContentScale.Crop,
+            modifier = parallaxModifier.fillMaxWidth().preferredHeight(278.dp)
+        )
+    }
+}
+
+@Composable
+private fun PlantHeaderActions(
+    onBackClicked: () -> Unit,
+    onShareClicked: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Row(
+        modifier = modifier.fillMaxSize().systemBarsPadding(top = true).padding(top = 12.dp),
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        val iconModifier = Modifier.sizeIn(maxWidth = 32.dp, maxHeight = 32.dp)
+            .drawBackground(color = Color.White, shape = CircleShape)
```

Comment:
Should this be white in dark theme also?

Reply:
Ofc! Using surface color. Thanks!